targets/TARGET_STM/TARGET_STM32F7/flash_api.c

```diff
@@ -130,6 +130,9 @@ int32_t flash_erase_sector(flash_t *obj, uint32_t address)
         status = -1;
     }
 
+    SCB_CleanInvalidateDCache();
```

Comment:
A ranged invalidate-only would be less disruptive than a full clean+invalidate.

The I cache invalidate should also be ranged too, but it seems that CMSIS doesn't have a ranged version of `SCB_InvalidateICache` for some reason, despite the M7 supporting it.

Feels like the lock should be reapplied first - as soon as possible?

Reply:
For  the D cache, I tried with this code : 
{
     int i = 0;
     uint32_t address_sector = FLASH_BASE;
     for(i=0;i<SectorId;i++){
         address_sector += GetSectorSize(i);
     }
        
     SCB_CleanInvalidateDCache_by_Addr(&address_sector, GetSectorSize(SectorId));
     SCB_InvalidateICache();
 }
Unfortunately, this was not working.
Then, I think the time consumed by SCB_CleanInvalidateDCache() is not so important compared to the memory erase.

Reply:
You should just need to do `SCB_CleanInvalidateDCache_by_Addr(data, size)`, except for the fact that the original parameters have been modified.

Reply:
Ah, that was for the program. I see what you mean now for the erase code - you needed to mess to try to find the sector start address.

Your code looks correct, except you have `&address_sector` in the invalidate where it should be `address_sector`.

Reply:
Thanks for insisting ! I pushed a new version including your remarks. I invalidate and clean only the concern sector. I have also added a function to get sector base address.
The rebase on master forced me to clean my code and the test is passed now...